packages/react-core/src/components/Select/Select.tsx

```diff
@@ -522,10 +537,12 @@ class Select extends React.Component<SelectProps & InjectedOuiaProps, SelectStat
                   {toggleIcon && <span className={css(styles.selectToggleIcon)}>{toggleIcon}</span>}
                   <span className={css(styles.selectToggleText)}>{placeholderText}</span>
                   {!isCheckboxSelectionBadgeHidden &&
-                    selections &&
-                    (Array.isArray(selections) && selections.length > 0) && (
+                    (customBadgeText !== null ||
+                      (selections && Array.isArray(selections) && selections.length > 0)) && (
```

Comment:
You can use the `hasAnySelections` variable instead of `Array.isArray(selections) && selections.length > 0`.